### ISSUE TITLE
Unreachable API

### ISSUE THREAD

**Issue by Rekyt** [EnquistLab/RTNRS#7]
Hey @bmaitner,

I'm unsure about what's causing it but I've got an error when accessing TNRS. I've got the last GitHub version of `TNRS`.

Is the API online? How to check the issue?

reprex:

``` r
TNRS::TNRS_version()
#> There appears to be a problem reaching the API.
#> NULL
```

<sup>Created on 2022-04-08 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.1.2 (2021-11-01)
#>  os       macOS High Sierra 10.13.6
#>  system   x86_64, darwin17.0
#>  ui       X11
#>  language (EN)
#>  collate  fr_FR.UTF-8
#>  ctype    fr_FR.UTF-8
#>  tz       Europe/Paris
#>  date     2022-04-08
#>  pandoc   2.11.2 @ /Applications/RStudio.app/Contents/MacOS/pandoc/ (via rmarkdown)
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version date (UTC) lib source
#>  cli           3.2.0   2022-02-14 [1] CRAN (R 4.1.2)
#>  crayon        1.5.1   2022-03-26 [1] CRAN (R 4.1.2)
#>  curl          4.3.2   2021-06-23 [1] CRAN (R 4.0.3)
#>  digest        0.6.29  2021-12-01 [1] CRAN (R 4.0.2)
#>  ellipsis      0.3.2   2021-04-29 [1] CRAN (R 4.0.3)
#>  evaluate      0.15    2022-02-18 [1] CRAN (R 4.1.2)
#>  fansi         1.0.3   2022-03-24 [1] CRAN (R 4.1.2)
#>  fastmap       1.1.0   2021-01-25 [1] CRAN (R 4.0.3)
#>  fs            1.5.2   2021-12-08 [1] CRAN (R 4.0.2)
#>  glue          1.6.2   2022-02-24 [1] CRAN (R 4.1.2)
#>  highr         0.9     2021-04-16 [1] CRAN (R 4.0.3)
#>  htmltools     0.5.2   2021-08-25 [1] CRAN (R 4.0.2)
#>  httr          1.4.2   2020-07-20 [1] CRAN (R 4.0.2)
#>  jsonlite      1.8.0   2022-02-22 [1] CRAN (R 4.1.2)
#>  knitr         1.38    2022-03-25 [1] CRAN (R 4.1.2)
#>  lifecycle     1.0.1   2021-09-24 [1] CRAN (R 4.0.2)
#>  magrittr      2.0.3   2022-03-30 [1] CRAN (R 4.1.2)
#>  pillar        1.7.0   2022-02-01 [1] CRAN (R 4.1.2)
#>  pkgconfig     2.0.3   2019-09-22 [1] CRAN (R 4.0.2)
#>  purrr         0.3.4   2020-04-17 [1] CRAN (R 4.0.2)
#>  R.cache       0.15.0  2021-04-30 [1] CRAN (R 4.0.2)
#>  R.methodsS3   1.8.1   2020-08-26 [1] CRAN (R 4.0.2)
#>  R.oo          1.24.0  2020-08-26 [1] CRAN (R 4.0.2)
#>  R.utils       2.11.0  2021-09-26 [1] CRAN (R 4.0.3)
#>  R6            2.5.1   2021-08-19 [1] CRAN (R 4.0.2)
#>  reprex        2.0.1   2021-08-05 [1] CRAN (R 4.0.2)
#>  rlang         1.0.2   2022-03-04 [1] CRAN (R 4.1.2)
#>  rmarkdown     2.13    2022-03-10 [1] CRAN (R 4.1.2)
#>  rstudioapi    0.13    2020-11-12 [1] CRAN (R 4.0.2)
#>  sessioninfo   1.2.2   2021-12-06 [1] CRAN (R 4.0.2)
#>  stringi       1.7.6   2021-11-29 [1] CRAN (R 4.0.2)
#>  stringr       1.4.0   2019-02-10 [1] CRAN (R 4.0.2)
#>  styler        1.7.0   2022-03-13 [1] CRAN (R 4.1.2)
#>  tibble        3.1.6   2021-11-07 [1] CRAN (R 4.0.2)
#>  TNRS          0.3.0   2022-04-07 [1] Github (EnquistLab/RTNRS@8958ab1)
#>  utf8          1.2.2   2021-07-24 [1] CRAN (R 4.0.3)
#>  vctrs         0.4.0   2022-03-30 [1] CRAN (R 4.1.2)
#>  withr         2.5.0   2022-03-03 [1] CRAN (R 4.1.2)
#>  xfun          0.30    2022-03-02 [1] CRAN (R 4.1.2)
#>  yaml          2.3.5   2022-02-21 [1] CRAN (R 4.1.2)
#> 
#>  [1] /Library/Frameworks/R.framework/Versions/4.1/Resources/library
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

</details>

**Comment by bmaitner** [comment]
Hey @Rekyt Is this still giving you problems? If so I'll dig into the dependencies and see if I have a different version that might be driving it, but as of now I just ran it and it ran fine.

That particular message is only shown when httr::POST throws an error when connecting, which suggests that indeed the server may have been down. In the event of a bad query, it would display information on the http status code.  @ojalaquellueva has been doing some server upgrades recently, but I'm not sure if any of those would have been impacting things today.  @ojalaquellueva , was the TNRS API down at all today? I'm wondering if we need to include a better way to check whether things are up and running in the R package.



``` r
TNRS::TNRS_version()
#>   db_version build_date code_version api_version
#> 1        4.2 2021-08-02        5.0.3
```

<sup>Created on 2022-04-08 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
#> - Session info ---------------------------------------------------------------
#>  setting  value
#>  version  R version 4.1.2 (2021-11-01)
#>  os       Windows 10 x64 (build 17763)
#>  system   x86_64, mingw32
#>  ui       RTerm
#>  language (EN)
#>  collate  English_United States.1252
#>  ctype    English_United States.1252
#>  tz       America/New_York
#>  date     2022-04-08
#>  pandoc   2.14.0.3 @ C:/Program Files/RStudio/bin/pandoc/ (via rmarkdown)
#> 
#> - Packages -------------------------------------------------------------------
#>  package     * version date (UTC) lib source
#>  cli           3.2.0   2022-02-14 [1] CRAN (R 4.1.3)
#>  crayon        1.5.0   2022-02-14 [1] CRAN (R 4.1.3)
#>  curl          4.3.2   2021-06-23 [1] CRAN (R 4.1.1)
#>  digest        0.6.28  2021-09-23 [1] CRAN (R 4.1.1)
#>  ellipsis      0.3.2   2021-04-29 [1] CRAN (R 4.1.1)
#>  evaluate      0.15    2022-02-18 [1] CRAN (R 4.1.3)
#>  fansi         1.0.2   2022-01-14 [1] CRAN (R 4.1.3)
#>  fastmap       1.1.0   2021-01-25 [1] CRAN (R 4.1.1)
#>  fs            1.5.2   2021-12-08 [1] CRAN (R 4.1.3)
#>  glue          1.6.2   2022-02-24 [1] CRAN (R 4.1.3)
#>  highr         0.9     2021-04-16 [1] CRAN (R 4.1.1)
#>  htmltools     0.5.2   2021-08-25 [1] CRAN (R 4.1.1)
#>  httr          1.4.2   2020-07-20 [1] CRAN (R 4.1.1)
#>  jsonlite      1.8.0   2022-02-22 [1] CRAN (R 4.1.3)
#>  knitr         1.37    2021-12-16 [1] CRAN (R 4.1.2)
#>  lifecycle     1.0.1   2021-09-24 [1] CRAN (R 4.1.1)
#>  magrittr      2.0.2   2022-01-26 [1] CRAN (R 4.1.3)
#>  pillar        1.7.0   2022-02-01 [1] CRAN (R 4.1.2)
#>  pkgconfig     2.0.3   2019-09-22 [1] CRAN (R 4.1.1)
#>  purrr         0.3.4   2020-04-17 [1] CRAN (R 4.1.1)
#>  R.cache       0.15.0  2021-04-30 [1] CRAN (R 4.1.3)
#>  R.methodsS3   1.8.1   2020-08-26 [1] CRAN (R 4.1.1)
#>  R.oo          1.24.0  2020-08-26 [1] CRAN (R 4.1.1)
#>  R.utils       2.11.0  2021-09-26 [1] CRAN (R 4.1.1)
#>  R6            2.5.1   2021-08-19 [1] CRAN (R 4.1.1)
#>  reprex        2.0.1   2021-08-05 [1] CRAN (R 4.1.3)
#>  rlang         1.0.2   2022-03-04 [1] CRAN (R 4.1.3)
#>  rmarkdown     2.13    2022-03-10 [1] CRAN (R 4.1.3)
#>  rstudioapi    0.13    2020-11-12 [1] CRAN (R 4.1.1)
#>  sessioninfo   1.2.2   2021-12-06 [1] CRAN (R 4.1.2)
#>  stringi       1.7.6   2021-11-29 [1] CRAN (R 4.1.2)
#>  stringr       1.4.0   2019-02-10 [1] CRAN (R 4.1.1)
#>  styler        1.7.0   2022-03-13 [1] CRAN (R 4.1.3)
#>  tibble        3.1.6   2021-11-07 [1] CRAN (R 4.1.3)
#>  TNRS          0.3.0   2022-04-08 [1] Github (EnquistLab/RTNRS@8958ab1)
#>  utf8          1.2.2   2021-07-24 [1] CRAN (R 4.1.1)
#>  vctrs         0.3.8   2021-04-29 [1] CRAN (R 4.1.1)
#>  withr         2.5.0   2022-03-03 [1] CRAN (R 4.1.3)
#>  xfun          0.28    2021-11-04 [1] CRAN (R 4.1.2)
#>  yaml          2.2.1   2020-02-01 [1] CRAN (R 4.1.1)
#> 
#>  [1] C:/Users/Public/R-4.1.2/library
#> 
#> ------------------------------------------------------------------------------
```

</details>

**Comment by Rekyt** [comment]
Hi Brian, thank you for your quick answer.
Something I didn't do is to test the query on a second computer and it works perfectly with both the CRAN and the GitHub versions of `TNRS`.

So it seems there may be an issue with my computer here... I don't know how to track it down though.

**Comment by Rekyt** [comment]
FYI, I'm investigating the error and this is waht I'm facing right now:
I've looked into `TNRS:::TNRS_core()` and the problem comes from the call to `POST()` indeed.

I'm getting the following error:

```
Erreur dans curl::curl_fetch_memory(url, handle = handle) : 
  SSL certificate problem: certificate has expired
 ```
 
 I'll continue to lookup to understand what causes this error in `curl`.

**Comment by Rekyt** [comment]
So, it clearly seems to be a problem from my side as the certificate is perfectly valid with: https://www.sslshopper.com/ssl-checker.html#hostname=https://tnrsapi.xyz/tnrs_api.php

So it seems an issue with curl.

I could temporarily avoid the issue removing the SSL check from `httr` using `httr::set_config(httr::config(ssl_verifypeer = 0L))` (as I found in this comment https://github.com/r-lib/pkgdown/issues/1157#issuecomment-538175656).

My understanding was that the version of `curl` I was using was not compatible with the new certificates. And I had a discrepancy between the system version of `curl` and the version installed through Homebrew. I had to follow [these instructions](https://github.com/jeroen/curl/#macos-using-curl-from-homebrew) to bypass the system version of curl and solve the problem.

Totally a problem from my side but good to now if other users have certificate issues!
